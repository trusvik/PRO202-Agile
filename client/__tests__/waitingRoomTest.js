// WaitingRoom.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import WaitingRoom from '../components/user/waitingRoom';

// Mock the `useNavigate` hook
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => jest.fn(),
}));

// Mock the WebSocket
global.WebSocket = jest.fn(() => ({
    onopen: jest.fn(),
    onmessage: jest.fn(),
    onclose: jest.fn(),
    send: jest.fn(),
    close: jest.fn(),
}));

describe('WaitingRoom Snapshot Test', () => {
    it('renders correctly', () => {
        sessionStorage.setItem('names', JSON.stringify(['Player1', 'Player2'])); // Mock session storage

        // Render the component
        const tree = renderer.create(
            <MemoryRouter>
                <WaitingRoom />
            </MemoryRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
