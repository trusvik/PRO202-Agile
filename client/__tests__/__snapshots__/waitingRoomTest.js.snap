// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WaitingRoom Snapshot Test renders correctly 1`] = `
<div
  className="waitingBody"
>
  <div
    className="roomContainer"
  >
    <h1>
      Venterom
    </h1>
    <div
      className="statusMessage"
    >
      <p>
        Spillet starter straks. Vennligst vent...
      </p>
    </div>
    <div
      className="nameList"
    />
  </div>
</div>
`;
